### PR TITLE
Plc1

### PR DESCRIPTION
**Pull request by daizu27**
Doing some more writing, here's a quick post on some PLC work I've been doing. @CamDavidsonPilon

### CODE REVIEW

File: _posts/2023-08-30-industrial-computers-for-koji.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+![](/assets/images/koji_plc1/testbench1.JPG){:width="450" .center}
+
+Here's where you can really appreciate the convenience of industrial control panels. Everything here is mounted on a DIN rail, a piece of steel channel 35mm wide that allows users to switch out any component as needed. All wires are terminated at screw-in terminal blocks, meaning things can be re-wired quickly without soldering, easily diagnosed with a multimeter, but still be secure enough not to come loose from vibrations. 
+
+On the back of the test bench, we'll mount our sensors. These would normally go into the 'field', in our case, the koji room. 
+
+![](/assets/images/koji_plc1/testbench2.JPG){:width="450" .center}
+
+<div class="callout">
+⚠ We are dealing with 120VAC line voltage, which can be dangerous. If you plan on DIY-ing something based on this post, please make sure you know what you're doing, or get a competent electrician. Also, there are codes that you must follow if you're actually putting together an industrial panel. This post is mostly for 'educational' purposes - my intent isn't to tell you how to put together a panel, but how PLCs work at a basic level. Stay safe!
```
Comment (CamDavidsonPilon) [inline _posts/2023-08-30-industrial-computers-for-koji.md:140]: 👍

File: _posts/2023-08-30-industrial-computers-for-koji.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+</div>
+
+
+## Next Steps and Notes
+
+That's it for now! I'll do some basic testing on this test bench before transferring it to a proper electrical enclosure, with some additional functionality. 
+
+I'll be using this PLC to control a "general purpose" solid-state fermentation room, in order to produce time-dependent temperature/RH/CO2 profiles for koji, natto, and mushroom growing. 
+
+This post barely scratches the surface of the capabilities of PLCs. I won't be going into any wiring details, electrical connections, how relays work, etc. If you want to learn more, see the 'References' section below.
```
Comment (CamDavidsonPilon) [inline _posts/2023-08-30-industrial-computers-for-koji.md:150]: I think I would like to know more about how you built this set up (speaking on behalf of my self and the audience). Is that for a future post?

**Review by CamDavidsonPilon — COMMENTED** [review]
Is the next step to build out a micro-koji room in your home?! 

I think this is a good post, and I've interested to learn more (lots of overlap with what I'm doing with Pioreactor tbh). I left a few comments on areas I think could be expanded on.

**Comment by CamDavidsonPilon** [comment]
feel free to merge when you feel its ready 👍

File: _posts/2023-08-30-industrial-computers-for-koji.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+![](/assets/images/koji_plc1/plc1.JPG){:width="450" .center}
+
+Bonus: programming software for Koyo Click is free!
+
+Thank you, Automation Direct (they are not a sponsor of this post). 
+
+Since this project is mostly 'for fun', we can also also get away with refurbished or used industrial components. Our target budget for this project will be around \$1200 CAD - like your average home PC build, and not out of reach for small scale koji/natto/mushroom makers. 
+
+After purchasing the necessary DIN rails, breakers, fuses, power supplies, terminal blocks, and relays, we can set up our system on a mock "test bench":
```
[the remark this answers is not in the document]
Comment (daizu27) [reply _posts/2023-08-30-industrial-computers-for-koji.md:127]: Added some notes at the end!

File: _posts/2023-08-30-industrial-computers-for-koji.md (hunk 2 above)
Comment (daizu27) [reply _posts/2023-08-30-industrial-computers-for-koji.md:148]: Hmm that might be a good idea, I might save it when I end up throwing this thing into panel

File: _posts/2023-08-30-industrial-computers-for-koji.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+## Building a PLC system
+
+Hopefully you're convinced by now, so let's put together a PLC system. 
+
+### Inputs and Outputs
+
+Our first system will be simple. The PLC will have the following I/O:
+
+1. **Analog Inputs (AI):** Room RH (4-20mA), Room CO2 (4-20mA), Room temperature (TC Type K), and Product Temperature (TC Type K).
```
[the remark this answers is not in the document]
Comment (daizu27) [reply _posts/2023-08-30-industrial-computers-for-koji.md]: Done

File: _posts/2023-08-30-industrial-computers-for-koji.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+That's it for now! I'll do some basic testing on this test bench before transferring it to a proper electrical enclosure, with some additional functionality. 
+
+I'll be using this PLC to control a "general purpose" solid-state fermentation room, in order to produce time-dependent temperature/RH/CO2 profiles for koji, natto, and mushroom growing. 
+
+This post barely scratches the surface of the capabilities of PLCs. I won't be going into any wiring details, electrical connections, how relays work, etc. If you want to learn more, see the 'References' section below. 
+
+A few final notes:
+
+- There are plenty of "budget" PLCs out there. I haven't worked with all of them, so I can't really offer my opinion on them. Certainly Koyo Click and Productivity Open, both offered by AutomationDirect, are the cheapest.
+- Having gone through this exercise, I'm not really sure what the best way of measuring temperature is for koji applications (TCs vs RTDs vs others). I might try multiple methods, just for fun.
```
[the remark this answers is not in the document]
Comment (daizu27) [reply _posts/2023-08-30-industrial-computers-for-koji.md]: It does seem like RTDs are more reliable in general, good to know those work great for you, i might give PT1000s a shot

File: _posts/2023-08-30-industrial-computers-for-koji.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+In short, I don't have the space to build a full size koji room, so we'll do the next best thing, and build the control panel. It doesn't really matter what size the actual koji room is - the control system will be the same.
+
+![](/assets/images/koji_plc1/Controlpanel.png){:width="450" .center}
+
+## What is a PLC, and why use one?
+
+A Programmable Logic Controller (PLC) is an industrial grade computer. If you already know how PLCs work, skip this section.
+
+Industry uses specialized computers called PLCs to control and automate all of their manufacturing and processes. These PLCs read and write signals to equipment and other pieces of instrumentation in the field (I/O) and can communicate to other PLCs and Supervisory Control and Data Acquisition (SCADA) systems.
```
[the remark this answers is not in the document]
Comment (daizu27) [reply _posts/2023-08-30-industrial-computers-for-koji.md]: Done

**Comment by daizu27** [comment]
My planned next few posts will go something like:
> Ladder Logic
> SCADA
> Final product

**Pull request merged**